lib/auth/helpers.go

```diff
@@ -436,6 +437,13 @@ func NewTestAuthServer(cfg TestAuthServerConfig) (*TestAuthServer, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	// Auth initialization is done (including creation/updating of all singleton
+	// configuration resources) so now we can start the cache.
+	if c, ok := srv.AuthServer.Cache.(*cache.Cache); ok {
+		if err := c.Start(); err != nil {
```

Comment:
Is there a corresponding Stop that should happen on shutdown?

Reply:
Hm, good point.

Looks like in the real impl the cache exits when the process exit context is done:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/service/service.go#L2214

But in tests we are using context.Background:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/auth/helpers.go#L235

Prior behavior was calling Start() as well:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/cache/cache.go#L918-L925

I can't find anywhere that actually calls `authServer.Cache.Close()`

Reply:
ah, it's called internally when the context is done:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/cache/cache.go#L1060-L1064

Reply:
I'll update the tests to clean that up properly

Reply:
fixed: [fd89faa](https://github.com/gravitational/teleport/pull/36216/commits/fd89faa8240b9c67440654628a9225039c486273)

Reply:
Nice detective work - thanks for the fix.